### ISSUE TITLE
@react-email/render causes compile errors, extension of issue #485

### ISSUE THREAD

**Issue by grmatthews** [resend/react-email#641]
### Describe the Bug

So I have narrowed down an issue reported earlier that could not be solved where I was getting compile errors related to @react-email/render.

I have successfully used the @react-email package, but only when @react-email/render is NOT imported into my project. Importing render causes a compile error, as per issue #485.

The workaround I have is as follows:
1. In my main project source folder, do `npm install @react-email/<any package except 'render'>`
2. Create `.jsx` page and use `@react-email/<component>` elements
3. Wrap the email content in a `<div>` with display:one
4. Get the inner HTML of the `<div>` -- as an alternative to using render()

Not sure if this provides any more clues.

Is render() doing anything more special than just grabbing the inner HTML as such?

I also noticed in the code I have managed to get working that the imports are like this:

```typescript
import {
    Body,
    Container,
    Column,
    Head,
    Heading,
    Html,
    Img,
    Link,
    Preview,
    Row,
    Section,
    Text,
} from "@react-email/components"
```

So, my question is how should I be importing the render method?

I was doing it as follows:

```typescript
import { render } from "@react-email/render"
```

Is that right?

### Which package is affected (leave empty if unsure)

@react-email/render

### Link to the code that reproduces this issue

https://pxp38m.csb.app/   -- works ok, cannot reproduce in sandbox

### To Reproduce

basically just include the following in your code and get a compile error

```ts
import { render } from "@react-email/render"
```

I copied the code from here: https://react.email/docs/integrations/sendgrid

### Expected Behavior

Be able to use `render()` to get the HTML for a component
Not have a compile error

### What's your node version? (if relevant)

18.13

**Comment by grmatthews** [comment]
Update: I have created a replacement for 'render' which seems to work for React 18

```
import ReactDOM from "react-dom/client"

const renderEmailComponent = (Component, componentProps = {}, callback) => {
    const offscreenContainer = document.createElement("div")

    // Temporarily override console.error
    const originalConsoleError = console.error
    console.error = (...args) => {
        if (
            args.length > 0 &&
            typeof args[0] === "string" &&
            args[0].includes("<div> cannot appear as a child of <html>")
        ) {
            // Suppress the specific warning
            return
        }
        originalConsoleError(...args)
    }

    const root = ReactDOM.createRoot(offscreenContainer)
    root.render(<Component {...componentProps} />)

    // Wait for the rendering to be committed
    setTimeout(() => {
        const html = offscreenContainer.innerHTML
        root.unmount()

        // Restore the original console.error function
        console.error = originalConsoleError

        callback(html)
    }, 0)
}

export { renderEmailComponent }
```

The client code to call this is:

```
<Button onClick={() => {
  renderEmailComponent(EmailTest, {}, (html) => {
    console.log("%crendered HTML", "color:lightgreen", html)
    })
  }}
>
  Email Test
</Button>
```

**Comment by gabrielmfern** [comment]
@grmatthews Are you using next to build and the resend SDK to send an email inside of Next by any chance?

**Comment by gabrielmfern** [comment]
Closing this one as not planned as it seems to be stale. If anyone else experiences this issue, with the latest versions, please let me know and I'll reopen it.
